vortex-expr/src/pruning/pruning_predicate.rs

```diff
@@ -0,0 +1,81 @@
+use std::fmt::Display;
+
+use vortex_array::aliases::hash_map::HashMap;
+use vortex_array::aliases::hash_set::HashSet;
+use vortex_array::stats::Stat;
+use vortex_array::{Array, ArrayRef};
+use vortex_error::{VortexExpect as _, VortexResult};
+
+use super::field_or_identity::FieldOrIdentity;
+use super::pruning_predicate_rewriter::convert_to_pruning_expression;
+use super::relation::Relation;
+use crate::{ExprRef, Literal, Scope};
+
+#[derive(Debug, Clone)]
+pub struct PruningPredicate {
+    expr: ExprRef,
+    required_stats: Relation<FieldOrIdentity, Stat>,
+}
+
+impl Display for PruningPredicate {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "PruningPredicate({}, {{{}}})",
+            self.expr, self.required_stats
+        )
+    }
+}
```

Comment:
I think we should probably kill this, seems like it's only used for tests and logging, in which case no reason we can't use the Debug impl

Reply:
Done.